### PR TITLE
SONARJNKNS-292 Autopopulate PR values

### PR DESCRIPTION
**Pull request by timja**
Tested using:

https://github.com/hmcts/spring-boot-template/pull/57
https://sonarcloud.io/project/issues?id=uk.gov.hmcts.reform%3Aspring-boot-template&pullRequest=57&resolved=false

image

### CODE REVIEW

**Comment by Rob-Doherty** [comment]
This will be really useful!

**Comment by timja** [comment]
@henryju could you review please?

**Comment by henryju** [comment]
We usually try to group the work on our products by sprints. I don't know when the next sprint on the Jenkins plugin will be scheduled. Do you have a strong need to get this merged early?

**Comment by timja** [comment]
currently we have 60+ apps overwriting the master report on every branch that it is being run on.
If this isn't going to be merged soon we can look to either fork or add the systems properties ourself.
But it would be great if it could be done sooner

**Comment by Rob-Doherty** [comment]
@henryju has there been any update on this? would be really useful for us!

**Comment by henryju** [comment]
A sprint is in the pipe, but not likely to start before a few weeks. In the meantime, feel free to build yourself the plugin with this change.

**Comment by timja** [comment]
any update on this @henryju ?

**Comment by henryju** [comment]
We are all busy with other topics right now. But don't worry, that's not forgotten.

**Comment by timja** [comment]
@henryju any chance you could take a look please?

**Comment by henryju** [comment]
We are doing a sprint on PR autodetection for many CIs, including Jenkins. But we decided to implement it directly in the SonarQube Scanner engine.
Still I will use this PR as a source of inspiration.
See https://jira.sonarsource.com/browse/SONAR-11853

File: src/main/java/hudson/plugins/sonar/SonarBuildWrapper.java [hunk 1]
```diff
@@ -134,6 +135,20 @@ public void setUp(Context context, Run<?, ?> build, FilePath workspace, Launcher
     if (!token.isEmpty()) {
       sb.append(", \"sonar.login\" : \"").append(escapeJson(token)).append("\"");
     }
+
+    // Add pull request specific variables if we're on a pull request
+    if (initialEnvironment.get("CHANGE_ID") != null) {
+      sb.append(", \"sonar.pullrequest.branch\" : \"").append(escapeJson(initialEnvironment.get("CHANGE_BRANCH"))).append("\"");
+      sb.append(", \"sonar.pullrequest.key\" : \"").append(escapeJson(initialEnvironment.get("CHANGE_ID"))).append("\"");
+
+      if (initialEnvironment.get("GITHUB_AUTH") != null) {
```
[the remark this answers is not in the document]
Comment (timja) [reply src/main/java/hudson/plugins/sonar/SonarBuildWrapper.java]: not sure, I'm not getting that variable anymore which is likely why I'm not getting anything posted on the PRs.

I have the sonarcloud application installed but still don't seem to get checks, not sure why.

[the remark this answers is not in the document]
Comment (henryju) [reply src/main/java/hudson/plugins/sonar/SonarBuildWrapper.java]: OK, I will not rely on this variable, and only look at CHANGE_URL and see if it starts by github.com

**Comment by henryju** [comment]
Will be implemented directly in SonarQube/SonarCloud.

**Pull request closed without merging**